eta/core/video.py

```diff
@@ -4139,6 +4137,16 @@ def run(self, inpath, outpath, verbose=False):
         else:
             in_opts = self._in_opts
 
+        # Automatically determine the starting number of the inpath for cases
+        # where it's a sequence pattern (e.g. %06d.jpg).  The default behavior
+        # from ffmpeg is to start at 0 and look in the range [0,4].  If the first
+        # matched pattern begins above 4 we want to explicitly set that.
+        if "-start_number" not in in_opts and is_supported_image_sequence(
+            inpath
+        ):
+            start_number = next(iter(etau.parse_pattern(inpath)), 0)
+            in_opts.extend(["-start_number", str(start_number)])
```

Comment:
The design of the `FFmpeg` class is that you provide `in_opts` and `out_opts` at class instantiation and can then call `run()` one or more times to use those parameters on multiple videos.

Given that, I think you should include `in_opts = in_opts.copy()` here if you are going to modify the options based on a particular call to `run()`. Maybe the user calls `run()` again with a non-image-sequence input, for example. They wouldn't want `-start_number` in that call.

Alternatively, you could push the responsibility of knowing that `-start_number` needs to be included to the caller of `FFmpeg(in_opts=[...])`.

The `FFmpeg` class here was intended to be a very low-level wrapper around the `ffmpeg` binary, very little magic happening without the caller explicitly requesting it. Although in this particular case, I'm onboard with a little magic to make `/path/to/frames/%06d.jpg` work better, if you want to put it here.

Reply:
The condition where the `is_supported_image_sequence` method is checked accounts for the scenario where an image sequence is *not* provided.  As-is the only time this code runs is explicitly when an image sequence has been provided and the user did not themselves specify a `-start_number`.  The "magic" is that in such cases we automatically determine a starting point, however this point may in fact still be the ffmpeg default of 0 (i.e. the default specified in `next`) which won't cause any unexpected side-effects given that it's the built-in default.  In all other scenarios this behavior will be skipped entirely.

Reply:
I get that, I'm just pointing out that the current implementation modifies either the `FFmpeg.DEFAULT_IN_OPTS` or `FFmpeg._in_opts` list in-place, doesn't it?

```py
ffmpeg = FFmpeg()

# Oops, permanently modifies `FFmpeg.DEFAULT_IN_OPTS`
ffmpeg.run("/path/to/frames/%06d.jpg", "/path/for/video1.mp4")

# Doesn't this wrongly include `start_number` in `in_opts`?
ffmpeg.run("/path/to/video.mov", "/path/for/video2.mp4")
```

Reply:
Ah I see what you're driving at.  I'll push up a fix that calls the `copy()` method that resolves this issue